src/foam/dao/SequenceNumberDAO.java

```diff
@@ -4,15 +4,25 @@
 import foam.core.FObject;
 import foam.core.X;
 import foam.core.PropertyInfo;
+import foam.mlang.MLang;
 import foam.mlang.sink.Max;
 
 public class SequenceNumberDAO
-  extends ProxyDAO
+    extends ProxyDAO
 {
   protected String property = "id";
-  protected int value = 1;
+  protected long value = 1;
   private PropertyInfo property_ = null;
 
+  public SequenceNumberDAO(DAO delegate) {
+    setDelegate(delegate);
+  }
+
+  public SequenceNumberDAO(DAO delegate, String property) {
```

Comment:
Reverse the order of the arguments so that if you have a deeply decorated DAO the property is closer to the SequenceNumberDAO constructor and the code is easier to read.